incubator/hnc/controllers/object_controller.go

```diff
@@ -102,63 +104,96 @@ func (r *ObjectReconciler) ReconcileNamespace(ctx context.Context, log logr.Logg
 // update deletes this object if it's an obsolete copy, and otherwise ensures it's been propagated
 // to any child namespaces.
 func (r *ObjectReconciler) update(ctx context.Context, log logr.Logger, inst *unstructured.Unstructured) error {
-	// Make sure this object is actually supposed to exist. If not, delete it. This should
-	// trigger one more reconciliation cycle and we'll get the children then.
-	if deleted, err := r.deleteObsolete(ctx, log, inst); deleted || err != nil {
+	// Make sure this object is correct and supposed to propagate. If not, delete it or mark it as
+	// modified. This should trigger one more reconciliation cycle and we'll get the children then.
+	if success, err := r.ensureCorrectAncestry(ctx, log, inst); !success || err != nil {
```

Comment:
Nit: the function's called ensureCorrect so maybe call the resulting variable `correct` as opposed to `success`. We're not checking whether some operation succeeded or failed (that's what `err`is for), just whether this object was supposed to be here or not.

Reply:
done